### DISCUSSION TITLE
Unable to upgrade to mailu 2024.06

### DISCUSSION THREAD

**Discussion by Pegoku** [Mailu/Mailu#3437] [User support]
## Environment & Version

- `Docker Compose version v2.20.2`
- Version: `2024.06`

If you are not using docker compose do not file any new issue here.
Kubernetes related issues belong to https://github.com/Mailu/helm-charts/issues
If you are not using docker compose or kubernetes, create a new thread on user support in [disussions](https://github.com/Mailu/Mailu/discussions/categories/user-support).
Non-bug reports (or bug reports that do not follow the template) are moved to [disussions](https://github.com/Mailu/Mailu/discussions).

## Description

I upgraded mailu from 2.2 to 2024.06, but I'm getting this error. 
```
mailu-webmail-1   | WARNING: Changed defaults (These config options have new default values):
mailu-webmail-1   | - 'skin'
mailu-webmail-1   | - 'smtp_port'
mailu-webmail-1   | - 'smtp_user'
mailu-webmail-1   | - 'smtp_pass'
mailu-webmail-1   | - 'jquery_ui_skin_map'
mailu-webmail-1   | Executing database schema update.
mailu-webmail-1   | Updating database schema (2021100300)... 
mailu-webmail-1   | [FAILED]
mailu-webmail-1   | ERROR: Error in DDL upgrade 2021100300: [1] no such table: contacts
mailu-webmail-1   | Initializing database
mailu-webmail-1   | Already initialized
mailu-webmail-1   | Upgrading database
mailu-webmail-1 exited with code 4
```

## Replication Steps

upgrade from 2.2 to 2024.06

## Observed behaviour

Recieved errors

## Expected behaviour

No errors on log and updates fine

**Comment by Pegoku** [comment]
Also get this error: 
```
mailu-oletools-1  | WARNING:root:Your CPU has Advanced Vector Extensions available, we recommend you enable hardened-malloc earlier in the boot process by adding LD_PRELOAD=/usr/lib/libhardened_malloc.so to your mailu.env
mailu-oletools-1  | <string>:63: SyntaxWarning: invalid escape sequence '\['
mailu-oletools-1  | <string>:204: DeprecationWarning: There is no current event loop
mailu-oletools-1  | CRITICAL:root:Received SIGTERM, terminating.
mailu-antispam-1  | 2024-09-09 01:04:49 #1(main) <13ry3x>; lua; multimap.lua:1396: added multimap rule: freemail_to (header)
mailu-oletools-1  | WARNING:root:Your CPU has Advanced Vector Extensions available, we recommend you enable hardened-malloc earlier in the boot process by adding LD_PRELOAD=/usr/lib/libhardened_malloc.so to your mailu.env
mailu-oletools-1  | <string>:63: SyntaxWarning: invalid escape sequence '\['
mailu-oletools-1  | <string>:204: DeprecationWarning: There is no current event loop
```

even though I have the LD_PRELOAD added to mailu.env

**Comment by Diman0** [comment]
Do you mean you upgraded from Mailu 2.0 instead of 2.2?
Did you use the latest version of 2.0 before updating?
The logging in oletools is harmless.

```
mailu-webmail-1   | Updating database schema (2021100300)... 
mailu-webmail-1   | [FAILED]
mailu-webmail-1   | ERROR: Error in DDL upgrade 2021100300: [1] no such table: contacts
```
This indicates the roundcube database is corrupt/incomplete. The roundcube database always contains the table contacts.
In Mailu 2024.06 the database is located at: `/mailu/webmail/roundcube.db`
You could remove it and restart Mailu to start with a clean Roundcube database.

**Comment by Pegoku** [comment]
will it remove my email addresses, or just the mails?
Also, I'm pretty sure it was `2.2`
```
  GNU nano 6.4                           docker-compose.yml.old                                    
# This file is auto-generated by the Mailu configuration wizard.
# Please read the documentation before attempting any change.
# Generated for compose flavor

version: '2.2'

services:
```

**Comment by Diman0** [comment]
Ah, that is the version used of the docker-compose yml. That has nothing to do with the mailu version.

Recreating the roundcoube database will not result in loss of emails, but only in user settings in roundcube such as a custom email address list.

**Comment by Pegoku** [comment]
ok, will try it now.

**Comment by Pegoku** [comment]
I'm still getting this error 
```
mailu-oletools-1  | WARNING:root:Your CPU has Advanced Vector Extensions available, we recommend you enable hardened-malloc earlier in the boot process by adding LD_PRELOAD=/usr/lib/libhardened_malloc.so to your mailu.env
```

and this one 
```
mailu-front-1     | CRITICAL:root:Can't reach http://mail.pegoku.com/.well-known/acme-challenge/testing!, please ensure it's fixed or change the TLS_FLAVOR.
```
The last one is probably bc I'm using a Reverse proxy (NPM).

**Comment by Diman0** [comment]
In that case you need to configure your reverse proxy to let through the letsencrypt challenge. 
https://mailu.io/2024.06/compose/setup.html#tls-certificates
Or you can configure cert as tls flavor and copy the certificates from the reverse proxy server to the mailu server (assuming that the reverse proxy terminates TLS).

**Comment by Pegoku** [comment]
I already fixed the certs problem, using the certbot-cloudflare, but I'm getting `502` on the website

**Comment by Pegoku** [comment]
@Diman0 Could you help me to find the cause of the issue?
Thanks

**Reply by Diman0** [reply]
HTTP 502 can indicate the front container received an error from the admin container.
You need to check the front container logs and the admin container logs to determine why you receive a 502 error. Can you see any errors in the logs?

**Reply by Pegoku** [reply]
The only thing I can see are 2 IPs trying to connect to the port 25. 
``` 
192.168.203.3 - - [09/Sep/2024:16:27:18 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-admin-1  | 192.168.203.3 - - [09/Sep/2024:16:27:26 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-admin-1  | 192.168.203.3 - - [09/Sep/2024:16:27:33 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-admin-1  | 192.168.203.3 - - [09/Sep/2024:16:27:48 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-front-1  | 2024/09/09 16:17:43 [info] 25#25: *122 client 80.94.95.215:52234 connected to 0.0.0.0:25
mailu-front-1  | 2024/09/09 16:17:50 [info] 25#25: *122 client login failed: "AUTH not supported" while in http auth state, client: 80.94.95.215, server: 0.0.0.0:25, login: "fujj@example.com"
mailu-front-1  | 2024/09/09 16:17:58 [info] 25#25: *126 client 80.94.95.215:55344 connected to 0.0.0.0:25
mailu-admin-1  | 192.168.203.3 - - [09/Sep/2024:16:27:54 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-admin-1  | 192.168.203.3 - - [09/Sep/2024:16:28:03 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-admin-1  | 192.168.203.3 - - [09/Sep/2024:16:28:17 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-admin-1  | 192.168.203.3 - - [09/Sep/2024:16:28:19 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-front-1  | 2024/09/09 16:18:04 [info] 25#25: *126 client login failed: "AUTH not supported" while in http auth state, client: 80.94.95.215, server: 0.0.0.0:25, login: "stats@example.com"
mailu-front-1  | 2024/09/09 16:18:04 [info] 25#25: *130 client 80.94.95.239:60238 connected to 0.0.0.0:25
mailu-front-1  | 2024/09/09 16:18:09 [info] 25#25: *130 client login failed: "AUTH not supported" while in http auth state, client: 80.94.95.239, server: 0.0.0.0:25, login: "madeline@example.com"
mailu-admin-1  | 192.168.203.3 - - [09/Sep/2024:16:28:32 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-admin-1  | 192.168.203.3 - - [09/Sep/2024:16:28:47 +0200] "GET /internal/auth/email HTTP/1.0" 200 0 "-" "-"
mailu-front-1  | 2024/09/09 16:18:12 [info] 25#25: *135 client 80.94.95.215:58366 connected to 0.0.0.0:25 
```
Should I block them?

**Reply by Pegoku** [reply]
Also found SendGrid IPs (eg `167.89.6.224`, `167.89.14.105`) Should I block them too?

**Reply by Pegoku** [reply]
I think I found the issue. The website port was `:80`, I had it set to `:81` and also tried with `:8080` bc on the logs it says ` Listening at: http://0.0.0.0:8080`.
What port should the site be?

**Comment by Pegoku** [comment]
Now I can't add the mail account to my android phone bc `Certificate is not valid`  It expires on 2023. How can I renew the cert?

**Reply by Pegoku** [reply]
got it working by editing the `/mailu/certs.sh`. In the wiki it uses `docker exec mailu_front_1 ...` but (atleast for me) it's `docker exec mailu-front-1...`
